clients/src/main/java/org/apache/kafka/clients/NodeApiVersions.java

```diff
@@ -91,19 +95,11 @@ public static NodeApiVersions create(short apiKey, short minVersion, short maxVe
                 .setMaxVersion(maxVersion)));
     }
 
-    public NodeApiVersions(ApiVersionCollection nodeApiVersions) {
-        for (ApiVersion nodeApiVersion : nodeApiVersions) {
-            if (ApiKeys.hasId(nodeApiVersion.apiKey())) {
-                ApiKeys nodeApiKey = ApiKeys.forId(nodeApiVersion.apiKey());
-                supportedVersions.put(nodeApiKey, nodeApiVersion);
-            } else {
-                // Newer brokers may support ApiKeys we don't know about
-                unknownApis.add(nodeApiVersion);
-            }
-        }
+    public NodeApiVersions(Collection<ApiVersion> nodeApiVersions) {
```

Comment:
Is it necessary to have a special constructor for the case where there are no supported features? This won't be a common situation in the future, right? This feels like the kind of thing that could easily lead to mistakes, if someone accidentally uses it outside of test code.

Looking at my IDE I see 10 uses of this in test code. Can we just change them to use the two-argument form?

Reply:
I think the constructors and factory methods on this class could use some cleanup, but it touches a lot of files so I was hoping to do this as a follow-up. For now, I'll remove the new constructor that was added